libraries/AC_AttitudeControl/AC_AttitudeControl.cpp

```diff
@@ -172,6 +172,42 @@ void AC_AttitudeControl::relax_attitude_controllers()
     reset_rate_controller_I_terms();
 }
 
+// Ensure attitude controller have zero errors to relax roll and yaw rate controller outputs
+void AC_AttitudeControl::relax_roll_yaw_controllers()
+{
+    // Get the current target attitude
+    Vector3f target_euler_angle;
+    _attitude_target_quat.to_euler(target_euler_angle.x, target_euler_angle.y, target_euler_angle.z);
+
+    // Get the current attitude quaternion and euler angles
+    Quaternion current_attitude;
+    _ahrs.get_quat_body_to_ned(current_attitude);
+    Vector3f current_euler_angle;
+    current_attitude.to_euler(current_euler_angle.x, current_euler_angle.y, current_euler_angle.z);
+
+    // Set the target attitude to the current roll and yaw attitude, retaining desired euler pitch
+    _attitude_target_quat.from_euler(current_euler_angle.x, target_euler_angle.y, current_euler_angle.z);
+
+    _attitude_ang_error.initialise();
+
+    // Initialize the angular rate variables to the current rate
+    _attitude_target_ang_vel = _ahrs.get_gyro();
+    ang_vel_to_euler_rate(_attitude_target_euler_angle, _attitude_target_ang_vel, _attitude_target_euler_rate);
+    _rate_target_ang_vel = _ahrs.get_gyro();
+
+    // Initialize remaining variables
+    _thrust_error_angle = 0.0f;
+
+    // Reset the PID filters
+    get_rate_roll_pid().reset_filter();
+    get_rate_pitch_pid().reset_filter();
```

Comment:
reset the pitch filter in a function called `relax_roll_yaw_controllers()` Maybe move it up to the relax_attitude_control() plane function to avoid confusion.

Reply:
I'm not sure whether the pitch filter should be reset here, or not; need to do some testing.